arangod/Aql/types.h

```diff
@@ -65,8 +70,28 @@ using AqlCollectionMap = std::map<std::string, aql::Collection*, std::less<>>;
 struct Variable;
 using VarSet = std::unordered_set<Variable const*>;
 using VarSetStack = std::vector<VarSet>;
-using RegIdSetStack = std::vector<std::unordered_set<RegisterId>>;
-using RegIdOrderedSetStack = std::vector<std::set<RegisterId>>;
+
+template<typename T>
+struct non_empty_vector : std::vector<T> {
+  non_empty_vector(std::initializer_list<T> list) : std::vector<T>(std::move(list)) {
+    TRI_ASSERT(!this->empty());
+  }
+
+  non_empty_vector() = default;
```

Comment:
Does it mean one can create an empty `non_empty_vector` via just `non_empty_vector lol;`?
Why the hell is this needed?

Reply:
this is just for debugging and will not stay. that's why this is a draft pr, it's not ready for review.